classes/Twig/UpdateSteps.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace PrestaShop\Module\AutoUpgrade\Twig;
+
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator;
+
+class UpdateSteps
+{
+    const VERSION_CHOICE = 'version-choice';
+    const UPDATE_OPTIONS = 'update-options';
+    const BACKUP = 'backup';
+    const UPDATE = 'update';
+    const POST_UPDATE = 'post-update';
+
+    const STATE_NORMAL = 'normal';
+    const STATE_CURRENT = 'current';
+    const STATE_DONE = 'done';
+
+    /** @var Translator */
+    private $translator;
+
+    /** @var array<string, array<string,string>> */
```

Comment:
I'm not sure this can work, it is worth a try.

```suggestion
    /** @var array<self::STEP_*, array<string,string>> */
```